portality/templates/doaj/widgets.html

```diff
@@ -1,138 +1,84 @@
+    if (typeof SEARCH_CONFIGURED_OPTIONS != 'undefined') {
+        widget_fv_opts["queryString"] = SEARCH_CONFIGURED_OPTIONS;
+        widget_fv_opts["rawQuery"] = true;
     }
-&lt;/script&gt;
-&lt;script src="{{ config.get('BASE_URL') }}/static/widget/fixed_query{{ widget_filename_suffix }}.js" type="text/javascript"&gt;&lt;/script&gt;
-&lt;div id="doaj-fixed-query-widget"&gt;&lt;/div&gt;
-</pre>
-<br>
-Example:
-
-{% if env != 'production' %}
-<div class="alert alert-danger"><b>Dev Note: </b>Make sure BASE_URL is set correctly so that you are actually seeing the widget integration in the correct environment. Otherwise you might see the live widget integrated into your local development version of DOAJ, or other weird combinations like that. The current value of BASE_URL is {{ config.get('BASE_URL') }}. This should match whatever is in your browser address bar.</div>
-{% endif %}
-
-<iframe src="{{ url_for("doaj.fqw_demo") }}" style="width: 100%; height: 650px; border: 0"></iframe>
-
-<br>
-<p>The widget can resize nicely to fit within available horizontal space. Use the <code>page_size</code> property to minimise its
-vertical requirement by reducing the number of results per page.
-Note: the vertical size can fluctuate slightly depending on what is being showed on each page.</p>
+    // Otherwise, use options provided.
+    else if (typeof QUERY_OPTIONS != 'undefined') {
+        widget_fv_opts["rawQuery"] = false;
+
+        if(QUERY_OPTIONS.hasOwnProperty('selector')){
+            widget_fv_opts['selector'] = QUERY_OPTIONS.selector;
+        }
+
+        widget_fv_opts[`initialParams`] = {};
+        if (QUERY_OPTIONS.hasOwnProperty('size')) {
+            widget_fv_opts['initialParams']['size'] = QUERY_OPTIONS.size;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('from')) {
+            widget_fv_opts['initialParams']['from'] = QUERY_OPTIONS.from;
+        }
+
+        widget_fv_opts[`queryString`] = {};
+        if (QUERY_OPTIONS.hasOwnProperty('query_string')) {
+            widget_fv_opts['queryString']['queryString'] = QUERY_OPTIONS.query_string;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('query_field')) {
+            widget_fv_opts['queryString']['defaultField'] = QUERY_OPTIONS.query_field;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('search_operator')) {
+            widget_fv_opts['queryString']['defaultOperator'] = QUERY_OPTIONS.search_operator;
+        }
+
+        widget_fv_opts['sortFilter'] = {}
+        if (QUERY_OPTIONS.hasOwnProperty('sort_field')) {
+            widget_fv_opts['sortFilter']['field'] = QUERY_OPTIONS.sort_field;
+        }
+        if (QUERY_OPTIONS.hasOwnProperty('sort_direction')) {
+            widget_fv_opts['sortFilter']['sort_direction'] = QUERY_OPTIONS.sort_direction;
+        }
+
+        if (QUERY_OPTIONS.hasOwnProperty('search_type')) {
+            widget_fv_opts['es_type'] = QUERY_OPTIONS.search_type;
+        }
```

Comment:
We should be able to get rid of this duplicated code from this file if we get rid of `widget_fv_opts` completely

Reply:
Yes, as mentioned in my inital comment this is only to make debugging easier.